libs/execution_context/ebpf_maps.c

```diff
@@ -1094,21 +1199,24 @@ _initialize_lru_entry(_Inout_ ebpf_core_lru_map_t* map, _Inout_ ebpf_lru_entry_t
 static void
 _uninitialize_lru_entry(_Inout_ ebpf_core_lru_map_t* map, _Inout_ ebpf_lru_entry_t* entry)
 {
-    ebpf_lock_state_t state = ebpf_lock_lock(&map->lock);
-    ebpf_lru_key_state_t key_state = _get_key_state(map, entry);
-    ebpf_assert(key_state == EBPF_LRU_KEY_HOT || key_state == EBPF_LRU_KEY_COLD);
+    for (size_t partition = 0; partition < map->partition_count; partition++) {
+        ebpf_lock_state_t state = ebpf_lock_lock(&map->partitions[partition].lock);
+        ebpf_lru_key_state_t key_state = _get_key_state(map, partition, entry);
+        ebpf_assert(
+            key_state == EBPF_LRU_KEY_HOT || key_state == EBPF_LRU_KEY_COLD || key_state == EBPF_LRU_KEY_UNINITIALIZED);
```

Comment:
could valid code uninitialize an UNINITIALIZED entry?

Reply:
Yes. If the entry was never inserted into a partitions hot list it will be marked as unintialized.